File: Android/Android_Component/Components/app/src/main/java/ramon/better/com/components/activity/MainActivity.java
package ramon.better.com.components.activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import ramon.better.com.components.R;

public class MainActivity extends AppCompatActivity {

    private Button launchMode;
    private Button intentGo;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.launch_mode).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, StandardActivity.class);
                startActivity(intent);
            }
        });

        findViewById(R.id.intent_btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, IntentActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Android/Basic/Core/FourComponent/app/src/main/java/ramon/lee/fourcomponent/activity/MainActivity.kt
package ramon.lee.fourcomponent.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import ramon.lee.fourcomponent.R
import ramon.lee.fourcomponent.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private var binding: ActivityMainBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding?.onClick = OnClick()
    }

    inner class OnClick {
        fun onClick(v: View) {
            when(v.id) {
                R.id.btn_life_cycle -> {
                    startActivity(Intent(this@MainActivity, LifeCycleActivity::class.java))
                }
                R.id.btn_launch_mode -> {
                    startActivity(Intent(this@MainActivity, StandardActivity::class.java))
                }
                R.id.btn_service_test -> {
                    startActivity(Intent(this@MainActivity, ServiceTestActivity::class.java))
                }
            }
        }
    }
}